src/Listener/ORM/MediaEventSubscriber.php

```diff
@@ -44,22 +45,24 @@ public function onClear(): void
         $this->rootCategories = null;
     }
 
-    protected function recomputeSingleEntityChangeSet(EventArgs $args): void
+    protected function recomputeSingleEntityChangeSet(LifecycleEventArgs $args): void
     {
+        \assert($args instanceof ORMLifecycleEventArgs);
+
         $em = $args->getEntityManager();
 
         $em->getUnitOfWork()->recomputeSingleEntityChangeSet(
-            $em->getClassMetadata(\get_class($args->getEntity())),
-            $args->getEntity()
+            $em->getClassMetadata(\get_class($args->getObject())),
+            $args->getObject()
         );
     }
 
-    protected function getMedia(EventArgs $args)
+    protected function getMedia(LifecycleEventArgs $args)
     {
-        $media = $args->getEntity();
+        $media = $args->getObject();
 
         if (!$media instanceof MediaInterface) {
-            return $media;
+            throw new \RuntimeException('There is no media on the persistence event.');
         }
 
         if ($this->container->has('sonata.media.manager.category') && !$media->getCategory()) {
```

Comment:
How could I fix this error? this method is on the Media model, but not on the MediaInterface (this was done when splitting media-bundle and classification-bundle to avoid having to install classification when only using media.

Reply:
You could have a check `method_exists` ; better would be a check `instanceOf SomeInterface`.

This split seems weird because I see `use Sonata\ClassificationBundle\Model\CategoryInterface;` in the `class Media`.
If I click on this, I end in the test/Fixtures...

Reply:
I just opened an issue for this, adding the method_exists in the meanwhile.